src/Graph/Trees/DecisionTree.php

```diff
@@ -468,26 +462,60 @@ protected function _exportGraphviz(
                 $this->_exportGraphviz($carry, $nodesCounter, $node->right(), $maxDepth, $featureNames, $thisNode, 2, $depth);
             }
         } elseif ($node instanceof Outcome) {
-            $carry .= "  N$thisNode [label=\"{$node->outcome()}";
+            $outcome = $node->outcome();
+            $impurity = $node->impurity();
 
-            if ($node->impurity() > 0.0) {
-                $carry .= "\\nImpurity={$node->impurity()}";
+            $carry .= "  N$thisNode [label=\"{$outcome}";
+
+            if ($impurity > 0.0) {
+                $carry .= "\\nImpurity={$impurity}";
             }
-            $carry .= '",style="rounded,filled",fillcolor=gray];' . PHP_EOL;
+
+            $carry .= '"';
+
+            if (is_string($outcome)) {
+                $hash = crc32($outcome) % self::MAX_COLORS;
+
+                $fillColor = substr('00000' . dechex($hash), -6);
```

Comment:
Now that I think about it we could also generate the hash as a hex string and then just truncate it beyond the first 24 bits